comictaggerlib/ctsettings/__init__.py

```diff
@@ -25,3 +34,115 @@
     "ct_ns",
     "group_for_plugin",
 ]
+
+
+class SettingsEncoder(json.JSONEncoder):
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, pathlib.Path):
+            return {"type": "pathlib.Path", "value": str(obj)}
+
+        # Let the base class default method raise the TypeError
+        return json.JSONEncoder.default(self, obj)
+
+
+class SettingsDecoder(json.JSONDecoder):
+    def __init__(self, definitions: settngs.Definitions, **kwargs):
+        kwargs["object_hook"] = self.object_hook
+        super().__init__(**kwargs)
+        self.definitions = definitions
+
+    def object_hook(self, obj):
+        for o, v in obj.items():
+            # The source name is required, use the object with the full file
+            if o.startswith("Source ") and isinstance(v, dict):
+                for key, value in v.items():
+                    if isinstance(value, dict) and value.get("type"):
+                        typ = self.definitions[o].v[key].type
+                        if typ:
+                            # Should check types match?
+                            obj[o][key] = typ(value["value"])
+
+        return obj
```

Comment:
After looking at it the JSON api doesn't provide an option that gives enough context. Instead it can just be processed after it's loaded from json.
eg
```python
def validate_types(options: settngs.Config[settngs.Values], definitions: settngs.Definitions) -> settngs.Config[settngs.Values]:
    # Go through each setting
    for group in definitions.values():
        for setting in group.v.values():
            # Get the value and if it is the default
            value, default = settngs.get_option(options.values, setting)
            if not default:
                if setting.type is not None:
                    # If it is not the default and the type attribute is not None
                    # use it to convert the loaded string into the expected value
                    if isinstance(value, str):
                        options.values[setting.group][setting.dest] = setting.type(value)
    return options
    ```